src/dune_pkg/opam_solver.ml

```diff
@@ -36,6 +36,13 @@ let opam_file_is_avoid_version (opam_file : OpamFile.OPAM.t) =
   List.mem opam_file.flags Pkgflag_AvoidVersion ~equal:Poly.equal
 ;;
 
+let opam_file_is_ocaml_package (opam_file : OpamFile.OPAM.t) =
+  Option.equal
+    OpamPackage.Name.equal
+    opam_file.name
+    (Some (OpamPackage.Name.of_string "ocaml"))
```

Comment:
Don't we have a conflict class for OCaml packages? Would be good to leave a comment why we can't use it here.

Reply:
I assume you're referring to the `ocaml-core-compiler` conflict class. The `ocaml` package isn't part of that class. It seems to only contain compiler implementations such as `ocaml-base-compiler` and `ocaml-variants`. How were you imagining to use this class to help here? Just as a way to avoid hard-coding the "ocaml" package name?

Reply:
Even if using the conflict class, we then would just hardcode one name (`ocaml-core-compiler`) for another (`ocaml`).

Reply:
It's not about hard coding one particular name over the other. It's about making sure our code stays consistent with recognizing the compiler. I recall that I hard coded the name of the compiler package in another feature and it turned out wrong. Another reason for doing this in a principled way is that dune should support compiler forks just as well as it supports the original compiler.